packs/chef/actions/lib/shell_out.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python2.7
+
+import sys
+import os
+import subprocess
+import select
+
+# Starts command in subprocess processing output as it appears.
+def shell_out(command, env={}):
+    env = os.environ.copy().update(env)
+    proc = subprocess.Popen(args=command, stdin=None, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, env=env)
+    while True:
+        fdlist = [proc.stdout.fileno(), proc.stderr.fileno()]
+        ios = select.select(fdlist, [], [])
+        for fd in ios[0]:
+            if fd == proc.stdout.fileno():
+                sys.stdout.write(proc.stdout.read())
```

Comment:
Thanks Kami for pointing out here, I wanted to use readline() it's a typo... This code is not so nice, maybe you might to know how to check how many bytes are available in the stream. It will be much better than readline() I suppose.